aio/tools/esm-loader/esm-loader.mjs

```
@@ -0,0 +1,54 @@
+import {pathToFileURL} from 'url';
+import resolveExports from './vendored/resolve.exports/index.js';
+
+/*
+  Custom module loader (see https://nodejs.org/api/cli.html#--experimental-loadermodule) to support
+  loading third-party packages in esm modules when the rules_nodejs linker is disabled. Resolves
+  third-party imports from the node_modules folder in the bazel workspace defined by 
+  process.env.NODE_MODULES_WORKSPACE_NAME, and uses the default resolver for all other imports.
+
+  This is required for esm scripts with third-party dependencies because rules_nodejs doesn't
+  patch imports of esm modules when the linker is disabled with --nobazel_run_linker.
+*/ 
+export async function resolve(specifier, context, defaultResolve) {
+    if (!isNodeOrNpmPackageImport(specifier)) {
+      return defaultResolve(specifier, context, defaultResolve);
+    }
+
+    const nodeModules = path.resolve('external', process.env.NODE_MODULES_WORKSPACE_NAME, 'node_modules');
+
+    const [packageName,] = getPackageAndPathFromSpecifier(specifier);
+    const pathToNodeModule = path.join(nodeModules, packageName);
+
+    const isInternalNodePackage = !fs.existsSync(pathToNodeModule);
+    if (isInternalNodePackage) {
+      return defaultResolve(specifier, context, defaultResolve);
+    }
+
+    const packageJson = JSON.parse(fs.readFileSync(path.join(pathToNodeModule, 'package.json'), "utf-8"));
+
+    const localPackagePath = resolvePackageLocalFilepath(specifier, packageJson);
+    const resolvedFilePath = path.join(pathToNodeModule, localPackagePath);
+
+    return {url: pathToFileURL(resolvedFilePath).href};
+}
+
+function isNodeOrNpmPackageImport(specifier) {
+  return !specifier.startsWith('./') && !specifier.startsWith('../') && !specifier.startsWith('node:') && !specifier.startsWith('file:');
+}
+
+function getPackageAndPathFromSpecifier(specifier) {
+  const [, packageName, path = ''] = /^((?:@[^/]+\/)?[^/]+)(?:\/(.+))?$/.exec(specifier) ?? [];
```

Should we throw an error if the package path is not resolvable?